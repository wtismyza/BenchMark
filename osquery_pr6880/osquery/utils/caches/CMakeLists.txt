# Copyright (c) 2014-present, The osquery authors
#
# This source code is licensed as defined by the LICENSE file found in the
# root directory of this source tree.
#
# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)

function(osqueryUtilsCachesMain)

  if(OSQUERY_BUILD_TESTS)
    add_subdirectory("tests")
  endif()

  generateOsqueryUtilsCachesLru()
endfunction()

function(generateOsqueryUtilsCachesLru)
  add_library(osquery_utils_caches_lru INTERFACE)

  set(public_header_files
    lru.h
    lru-impl.h
  )

  generateIncludeNamespace(osquery_utils_caches_lru "osquery/utils/caches" "FILE_ONLY" ${public_header_files})

  add_test(NAME osquery_utils_caches_tests_lrutests-test COMMAND osquery_utils_caches_tests_lrutests-test)
endfunction()

osqueryUtilsCachesMain()
