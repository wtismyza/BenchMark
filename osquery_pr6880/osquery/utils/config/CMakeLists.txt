# Copyright (c) 2014-present, The osquery authors
#
# This source code is licensed as defined by the LICENSE file found in the
# root directory of this source tree.
#
# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)

function(osqueryUtilsConfigMain)
  generateOsqueryUtilsConfig()
endfunction()

function(generateOsqueryUtilsConfig)
  add_osquery_library(osquery_utils_config EXCLUDE_FROM_ALL
    default_paths.cpp
  )

  target_link_libraries(osquery_utils_config PUBLIC osquery_cxx_settings)

  set(public_header_files
    default_paths.h
  )

  generateIncludeNamespace(osquery_utils_config "osquery/utils/config" "FILE_ONLY" ${public_header_files})
endfunction()

osqueryUtilsConfigMain()
