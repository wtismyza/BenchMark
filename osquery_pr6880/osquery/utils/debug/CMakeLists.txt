# Copyright (c) 2014-present, The osquery authors
#
# This source code is licensed as defined by the LICENSE file found in the
# root directory of this source tree.
#
# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)

function(osqueryUtilsDebugMain)

  if(OSQUERY_BUILD_TESTS)
    generateOsqueryUtilsDebugDebugtestsTest()
  endif()

  generateOsqueryUtilsDebug()
endfunction()

function(generateOsqueryUtilsDebug)
  add_osquery_library(osquery_utils_debug INTERFACE)

  target_link_libraries(osquery_utils_debug INTERFACE
    osquery_cxx_settings
    thirdparty_boost
  )

  set(public_header_files
    debug_only.h
  )

  generateIncludeNamespace(osquery_utils_debug "osquery/utils/debug" "FILE_ONLY" ${public_header_files})

  add_test(NAME osquery_utils_debug_debugtests-test COMMAND osquery_utils_debug_debugtests-test)

endfunction()

function(generateOsqueryUtilsDebugDebugtestsTest)
  add_osquery_executable(osquery_utils_debug_debugtests-test tests/debug_only.cpp)

  target_link_libraries(osquery_utils_debug_debugtests-test PRIVATE
    osquery_cxx_settings
    osquery_utils_debug
    thirdparty_googletest
  )
endfunction()

osqueryUtilsDebugMain()
