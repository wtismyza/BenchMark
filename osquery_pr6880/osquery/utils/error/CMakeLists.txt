# Copyright (c) 2014-present, The osquery authors
#
# This source code is licensed as defined by the LICENSE file found in the
# root directory of this source tree.
#
# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)

function(osqueryUtilsErrorMain)

  if(OSQUERY_BUILD_TESTS)
    generateOsqueryUtilsErrorErrortestsTest()
  endif()

  generateOsqueryUtilsError()
endfunction()

function(generateOsqueryUtilsError)
  add_osquery_library(osquery_utils_error INTERFACE)

  target_link_libraries(osquery_utils_error INTERFACE
    osquery_cxx_settings
    osquery_utils_conversions_to
    osquery_utils_attribute
  )

  set(public_header_files
    error.h
  )

  generateIncludeNamespace(osquery_utils_error "osquery/utils/error" "FILE_ONLY" ${public_header_files})

  add_test(NAME osquery_utils_error_errortests-test COMMAND osquery_utils_error_errortests-test)

endfunction()

function(generateOsqueryUtilsErrorErrortestsTest)
  add_osquery_executable(osquery_utils_error_errortests-test tests/error.cpp)

  target_link_libraries(osquery_utils_error_errortests-test PRIVATE
    osquery_cxx_settings
    osquery_utils_error
    thirdparty_googletest
  )
endfunction()

osqueryUtilsErrorMain()
