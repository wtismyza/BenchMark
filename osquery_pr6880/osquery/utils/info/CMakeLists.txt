# Copyright (c) 2014-present, The osquery authors
#
# This source code is licensed as defined by the LICENSE file found in the
# root directory of this source tree.
#
# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)

function(osqueryUtilsInfoMain)
  generateOsqueryUtilsInfo()
endfunction()

function(generateOsqueryUtilsInfo)
  add_osquery_library(osquery_utils_info EXCLUDE_FROM_ALL
    platform_type.cpp
    tool_type.cpp
    version.cpp
  )

  target_link_libraries(osquery_utils_info PUBLIC
    osquery_cxx_settings
    osquery_utils_conversions
    osquery_utils_macros
  )

  target_compile_definitions(osquery_utils_info PUBLIC
    OSQUERY_VERSION=${OSQUERY_VERSION_INTERNAL}
    OSQUERY_BUILD_VERSION=${OSQUERY_VERSION_INTERNAL}
    OSQUERY_BUILD_SDK_VERSION=${OSQUERY_VERSION_INTERNAL}
  )

  set(public_header_files
    platform_type.h
    tool_type.h
    version.h
  )

  generateIncludeNamespace(osquery_utils_info "osquery/utils/info" "FILE_ONLY" ${public_header_files})

endfunction()

osqueryUtilsInfoMain()
