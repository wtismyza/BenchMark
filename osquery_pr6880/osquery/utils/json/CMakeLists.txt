# Copyright (c) 2014-present, The osquery authors
#
# This source code is licensed as defined by the LICENSE file found in the
# root directory of this source tree.
#
# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)

function(osqueryUtilsJsonMain)

  if(OSQUERY_BUILD_TESTS)
    generateOsqueryUtilsJsonJsontestsTest()
  endif()

  generateOsqueryUtilsJson()
endfunction()

function(generateOsqueryUtilsJson)
  add_osquery_library(osquery_utils_json EXCLUDE_FROM_ALL
    json.cpp
  )

  target_link_libraries(osquery_utils_json PUBLIC
    osquery_cxx_settings
    osquery_utils
    osquery_utils_conversions
    osquery_utils_status
    osquery_utils_system_systemutils
    thirdparty_rapidjson
  )

  target_compile_definitions(osquery_utils_json PUBLIC
    RAPIDJSON_HAS_STDSTRING=1
  )

  set(public_header_files
    json.h
  )

  generateIncludeNamespace(osquery_utils_json "osquery/utils/json" "FILE_ONLY" ${public_header_files})

  add_test(NAME osquery_utils_json_jsontests-test COMMAND osquery_utils_json_jsontests-test)

endfunction()

function(generateOsqueryUtilsJsonJsontestsTest)
  add_osquery_executable(osquery_utils_json_jsontests-test tests/json.cpp)

  target_link_libraries(osquery_utils_json_jsontests-test PRIVATE
    osquery_cxx_settings
    osquery_utils_json
    thirdparty_googletest
  )
endfunction()

osqueryUtilsJsonMain()
