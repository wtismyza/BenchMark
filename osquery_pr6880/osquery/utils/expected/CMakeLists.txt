# Copyright (c) 2014-present, The osquery authors
#
# This source code is licensed as defined by the LICENSE file found in the
# root directory of this source tree.
#
# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)

function(osqueryUtilsExpectedMain)
  if(OSQUERY_BUILD_TESTS)
    generateOsqueryUtilsExpectedExpectedtestsTest()
  endif()

  generateOsqueryUtilsExpected()
endfunction()

function(generateOsqueryUtilsExpected)
  add_osquery_library(osquery_utils_expected INTERFACE)

  target_link_libraries(osquery_utils_expected INTERFACE
    osquery_cxx_settings
    osquery_utils_debug
    osquery_utils_error
    thirdparty_boost
  )

  set(public_header_files
    expected.h
  )

  generateIncludeNamespace(osquery_utils_expected "osquery/utils/expected" "FILE_ONLY" ${public_header_files})

  add_test(NAME osquery_utils_expected_expectedtests-test COMMAND osquery_utils_expected_expectedtests-test)

endfunction()

function(generateOsqueryUtilsExpectedExpectedtestsTest)
  add_osquery_executable(osquery_utils_expected_expectedtests-test tests/expected.cpp)

  target_link_libraries(osquery_utils_expected_expectedtests-test PRIVATE
    osquery_cxx_settings
    osquery_utils_expected
    thirdparty_googletest
  )
endfunction()

osqueryUtilsExpectedMain()
